code/workspaces/infrastructure-api/src/kubernetes/serviceApi.js

```diff
@@ -6,6 +6,8 @@ import { handleCreateError, handleDeleteError } from './core';
 
 const API_BASE = config.get('kubernetesApi');
 
+export const getHeadlessServiceName = name => `${name}-headless`;
```

Comment:
I don't think this will work if we change the service name so probably the stack builder will need to do this bit. Also it keeps this as a pure k8s API.